src/OpenTelemetry.Extensions.Hosting/README.md

```diff
@@ -79,12 +83,10 @@ public static class MyLibraryExtensions
                 "MyFeature requires a hosting TracerProviderBuilder instance."))
             .AddHostedService<MyHostedService>()
             .AddSingleton<MyService>()
-            .AddSingleton<MyProcessor>()
-            .AddSingleton<MySampler>();
+            .AddSingleton<BaseProcessor<Activity>>(new MyProcessor());
```

Comment:
This signature isn't very nice, maybe we should add an extension in hosting something like...

```csharp
        public static IServiceCollection AddActivityProcessor<T>(this IServiceCollection services, ServiceLifetime serviceLifetime = ServiceLifetime.Singleton)
            where T : BaseProcessor<Activity>
        {
            services.Add(ServiceDescriptor.Describe(typeof(BaseProcessor<Activity>), typeof(T), serviceLifetime));
            return services;
        }
```

Reply:
yes, agree with that.